packages/plugin/src/interpreter.ts

```diff
@@ -17,6 +17,18 @@ import { FieldInfoGenerator } from "./code-gen/field-info-generator";
 import {OurFileOptions, OurServiceOptions, readOurFileOptions, readOurServiceOptions} from "./our-options";
 
 
+const escapeCharacter = "$";
+export const reservedObjectProperties = new Set(["__proto__", "toString"]);
+export const reservedClassProperties = new Set([
+    // js built in
+    "__proto__", "toString", "name", "constructor",
+    // generic clients
+    "methods", "typeName", "options", "_transport",
+    // @grpc/grpc-js clients
+    "close", "getChannel", "waitForReady", "makeUnaryRequest", "makeClientStreamRequest", "makeServerStreamRequest", "makeBidiStreamRequest"
```

Comment:
Good call, thanks for adding those.

Reply:
You already had these 🙂 I just moved them out of the class methods and turned them into Sets for O(1) look-ups (and to avoid extra GC every time they got created). I also considered doing the same treatment for these, but decided it was even more outside the scope of this PR. https://github.com/timostamm/protobuf-ts/blob/44b0fe4b7fe202c4d8a51f785c60ce71ca717622/packages/plugin/src/code-gen/local-type-name.ts#L8-L9